File: 0046-permutations/0046-permutations.py
class Solution:
    def permute(self, nums: List[int]) -> List[List[int]]:
        if len(nums) == 1:
            return [nums]
        ans = []
        for i, n in enumerate(nums):
            lCopy = nums.copy()
            ext = lCopy.pop(i)
            for p in self.permute(lCopy):
                ans.append([ext] + p)
        return ans

File: 0046-permutations/0046-permutations.js
/**
 * @param {number[]} nums
 * @return {number[][]}
 */
var permute = function(nums) {
    if (nums.length === 1) return [nums];

    let ans = [];
    nums.forEach((n, i) => {
        const copy = nums.slice();
        const extracted = copy.splice(i, 1);
        permute(copy).forEach(p => {
            ans.push([extracted, ...p]);
        })
    });
    
    return ans;
};